liquibase-core/src/main/java/liquibase/change/core/ExecuteShellCommandChange.java

```diff
@@ -384,13 +395,32 @@ public void finish() {
             this.processStream = null;
 
             try {
-                StreamUtil.copy(processStream, outputStream);
+                copy(processStream, outputStream);
             } catch (IOException e) {
                 e.printStackTrace();
             }
 
         }
 
+        public void copy(InputStream inputStream, OutputStream outputStream) throws IOException {
+            Integer maxToCopy = getMaxStreamGobblerOutput();
+            byte[] bytes = new byte[1024];
+            int r = inputStream.read(bytes);
+            while (r > 0) {
+                if (maxToCopy != null && copiedSize > maxToCopy) {
+                    if (!loggedTruncated) {
+                        outputStream.write("...[TRUNCATED]...".getBytes());
+                        loggedTruncated = true;
```

Comment:
Should we put a `break` here ?

Reply:
Will we read to the end of file in this case ?

Reply:
Since the inputStream.read() is outside the if/else we will continue reading through the inputstream. I think we  have to make sure we do that since it is coming from the process and processeses will sometimes wait until the output is read before continuing/finishing.

Reply:
Really didn't know about it that another  thread can wait for this. I only knew that if run/call method ends the thread and no wait or sleep or some lock inside it finishes its task.  Sounds very interesting. 
Do you have some sources that discribe this situation ?

Reply:
article is more about reading files.
we just read stream.
blocking is 'default' Stream's read() behavior
```
public int read(byte[] b)
         throws IOException
Reads some number of bytes from the input stream and stores them into the buffer array b. 
The number of bytes actually read is returned as an integer. 
This method blocks until input data is available, end of file is detected, or an exception is thrown.
If the length of b is zero, then no bytes are read and 0 is returned; 
otherwise, there is an attempt to read at least one byte. If no byte is available because the stream is at the end of the file, the value -1 is returned; 
otherwise, at least one byte is read and stored into b.
The first byte read is stored into element b[0], the next one into b[1], and so on. 
The number of bytes read is, at most, equal to the length of b. 
Let k be the number of bytes actually read; these bytes will be stored in elements b[0] through b[k-1], leaving elements b[k] through b[b.length-1] unaffected.
The read(b) method for class InputStream has the same effect as:
 read(b, 0, b.length) 
Parameters:
    b - the buffer into which the data is read.
Returns:
    the total number of bytes read into the buffer, or -1 if there is no more data because the end of the stream has been reached.
```